pkg/asset/machines/aws/awsmachines.go

```diff
@@ -46,24 +46,29 @@ func GenerateMachines(clusterID string, in *MachineInput) ([]*asset.RuntimeFile,
 	var result []*asset.RuntimeFile
 
 	for idx := int64(0); idx < total; idx++ {
-		var subnet *capa.AWSResourceReference
-		// By not setting subnets for the machine, we let CAPA choose one for us
+		subnet := &capa.AWSResourceReference{}
+		zone := mpool.Zones[int(idx)%len(mpool.Zones)]
```

Comment:
is it safe?

Reply:
The only risk here is if `len(mpool.Zones) == 0`. Can that ever happen?

Reply:
iiuc it will never happen: the zones seems to be filled when empty for Control Plane nodes, and bootstrap - the only supported machines with CAPI at this moment: 
https://github.com/openshift/installer/blob/2f2847be1b81526b6128a93e4e643e615c2ea71c/pkg/asset/machines/clusterapi.go#L116-L129

Reply:
I think the only risk would be a new region with no availability zones.